amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTaker.java

```diff
@@ -401,12 +428,15 @@ private void updateAllLeases(Callable<Long> timeProvider)
                         leasesToTake.size());
             }
 
-            scope.addData("TotalLeases", numLeases, StandardUnit.COUNT, MetricsLevel.DETAILED);
+        } finally {
             scope.addData("ExpiredLeases", originalExpiredLeasesSize, StandardUnit.COUNT, MetricsLevel.SUMMARY);
```

Comment:
ExpiredLeases will not be set to appropriate value in the case of very old leases.